src/topp/QualityControl.cpp

```diff
@@ -420,7 +430,47 @@ class TOPPQualityControl : public TOPPBase
 
       if (export_evidence.isValid())
       {
-        export_evidence.exportFeatureMap(*fmap,cmap);
+        auto& cmapProtIds = cmap.getProteinIdentifications();
+        map<String,String> fastaMap {};
+        //map the identifier to the description so that we can access the description via the cmap-identifier
+        for(const auto& entry : protDescription)
+        {
+          fastaMap.emplace(entry.identifier, entry.description);
+        }
+
+        if(cmapProtIds.empty())
+        {
+          OPENMS_LOG_ERROR << "No ProteinIdentifications in consensusmap... aboarting" << std::endl;
+        }
+
+        // THIS IS OUR FALLBACK IF THE USER PROVIDED NO .fasta
+        if(protDescription.empty())
+        {
+          OPENMS_LOG_INFO << "No FASTA passed, looking for the default search parameters in consensusXML" << std::endl;
+          const String& fileName = cmapProtIds[0].getSearchParameters().db;
+          FASTAFile().load(fileName, protDescription);
+        }
+
+        OPENMS_LOG_INFO << "Annotating description to hits..." << std::endl;
+        //check for existing description in hit & only if that doesnt exist -> extract it from fasta
+        for(auto& protId : cmapProtIds)
+        {
+          auto& cmapProtHits = protId.getHits();
+          for(auto& hit : cmapProtHits)
+          {
+            //check if hit has alrdy a description...
+            if(hit.getDescription().empty())
+            {
+              //set the additional description from the corresponding fasta entry
+              hit.setDescription(fastaMap[hit.getAccession()]);
```

Comment:
Lucky you: if the FASTA file did not contain the accession, then `fastaMap[hit.getAccession()]` will create a new key/value pair in the map and return an empty string.
So nothing bad happens, but it would be nice to count these cases, e.g.
```
  auto it_acc_description = fasta_map.find(hit.getAccession());
  if (it_acc_description == fasta_map.end())
  {
     ++count_accessions_not_found;
     if (count_accessions_not_found < 10) LOG_WARN << "Protein with accession '" << hit.getAccession() << "' not found in FASTA file '" << fasta_filename << std::endl;

   }
   else
   {
     hit.setDescription(it->second);
   }
```

and then outside the loops:
```
if (count_accessions_not_found > 0)
{
   LOG_WARN << "Warning: " << count_accessions_not_found << " protein accessions were not found in FASTA file. Txt output will have missing information on protein names and possibly gene names." << std::endl;
}

Reply:
Ive implemented this counter and if i have done evereything correct, there are 0 accessions which have not occured.